verify/verify.go

```diff
@@ -92,7 +92,7 @@ func (db *DB) VerifySignatures(s *data.Signed, role string) error {
 	// Verify that a threshold of keys signed the data. Since keys can have
```

Comment:
While this section is getting some attention: I just found that this early return will cause the threshold signature check to fail if any single signature is invalid, even if there is a threshold of valid signatures. To fix, we should replace `return ErrInvalid` with `continue`.

```go
if err := verifier.Verify(msg, sig.Signature); err != nil {
	return ErrInvalid
}
```

Reply:
Yes, please refer to the [spec](https://theupdateframework.github.io/specification/latest/index.html) ("a threshold of keys" to prevent arbitrary software attacks) and [python-tuf](https://github.com/theupdateframework/python-tuf/blob/7a760691c69bb4d58d37245d9b5132ee91f20487/tuf/api/metadata.py#L438-L452).

Reply:
And let's add unit tests to make sure this mistake gets caught.

Reply:
Decided not to fix right now because it will break: (1) tests that expect the broken behaviour, and more importantly, (2) clients that expect this same behaviour. Will create another issue and track it there.